mlrun/datastore/azure_blob.py

```diff
@@ -29,47 +30,148 @@
+            import azure  # noqa
+        except ImportError as exc:
+            raise ImportError("Azure not installed") from exc
+
+        if not self._service_client:
+            self._do_connect()
+        return self._service_client
+
+    def _do_connect(self):
+        """
+        Connect to the BlobServiceClient, using user-specified connection details.
+        Tries connection string first, then credential, then account key, SAS token, and finally anonymous login.
+        Raises MLRunInvalidArgumentError if none of the connection details are available
+        based on do_connect in AzureBlobFileSystem:
+        https://github.com/fsspec/adlfs/blob/2023.9.0/adlfs/spec.py#L422
+        """
+        from azure.identity import ClientSecretCredential
+
+        storage_options = self.storage_options
+        credential_from_client_id = None
+        if (
+            storage_options.get("credential") is None
+            and storage_options.get("account_key") is None
+            and storage_options.get("sas_token") is None
+            and storage_options.get("client_id") is not None
+        ):
+            credential_from_client_id = ClientSecretCredential(
+                tenant_id=storage_options.get("tenant_id"),
+                client_id=storage_options.get("client_id"),
+                client_secret=storage_options.get("client_secret"),
+            )
+        try:
+            if storage_options.get("connection_string") is not None:
+                self._service_client = BlobServiceClient.from_connection_string(
+                    conn_str=storage_options["connection_string"],
+                    max_block_size=self.max_blocksize,
+                    max_single_put_size=self.max_single_put_size,
+                )
+            elif storage_options.get("account_name") is not None:
+                account_url: str = (
+                    f"https://{storage_options['account_name']}.blob.core.windows.net"
+                )
+
+                creds = [
+                    credential_from_client_id or storage_options.get("credential"),
+                    storage_options.get("account_key"),
+                ]
+                if any(creds):
+                    self._service_client = [
+                        BlobServiceClient(
+                            account_url=account_url,
+                            credential=cred,
+                            _location_mode="primary",
+                            max_block_size=self.max_blocksize,
+                            max_single_put_size=self.max_single_put_size,
+                        )
+                        for cred in creds
+                        if cred is not None
+                    ][0]
+                elif storage_options.get("sas_token") is not None:
+                    sas_token = storage_options.get("sas_token")
+                    if not sas_token.startswith("?"):
+                        sas_token = f"?{storage_options.get('sas_token')}"
```

Comment:
```suggestion
                        sas_token = f"?{sas_token}"
```

Reply:
Or 
```suggestion
                        sas_token = "?" + sas_token
```
if you prefer.